app/controllers/flowchart_node_controller.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+class FlowchartNodeController < ApplicationController
+  def create
+    prev_id = params[:prev_id]
+    is_child = params[:is_child]
+    begin
+      prev_node = FlowchartNode.find(prev_id)
+    rescue ActiveRecord::RecordNotFound
+      render status: 404, json: { error: "No node found with id #{prev_id}." }
+    else
+      ActiveRecord::Base.transaction do
+        @new_node = FlowchartNode.create(
+          text: params[:text],
+          header: params[:header],
+          button_text: params[:button_text],
+          next_question: params[:next_question],
+          is_root: false,
+          flowchart_id: prev_node.flowchart_id
+        )
+        if is_child == 'true'
+          @new_node.child_id = prev_node.child_id
+          prev_node.child_id = @new_node.id
+        else
+          @new_node.sibling_id = prev_node.sibling_id
+          prev_node.sibling_id = @new_node.id
+        end
+        @new_node.save!
+        prev_node.save!
+      end
+      render json: @new_node
+    end
+  end
+
+  def show
+    id = params[:id]
+    begin
+      node = FlowchartNode.find(id)
+    rescue ActiveRecord::RecordNotFound
+      render status: 404, json: { error: "No node found with id #{id}." }
+    else
+      render json: node
+    end
+  end
+
+  def update
+    id = params[:id]
+    begin
+      flowchart_node = FlowchartNode.update(
+        id,
+        text: params[:text],
+        header: params[:header],
+        button_text: params[:button_text],
+        next_question: params[:next_question]
+      )
+    rescue StandardError
+      render status: 404, json: { error: "No node found with id #{id}." }
+    else
+      render json: flowchart_node
+    end
+  end
+
+  def swap
+    id_a = params[:id_a]
+    id_b = params[:id_b]
+    begin
+      node_a = FlowchartNode.find(id_a)
+      node_b = FlowchartNode.find(id_b)
+    rescue ActiveRecord::RecordNotFound
+      render status: 404, json: { error: 'Error finding nodes with the given ids.' }
+    else
+      node_a.text, node_b.text = node_b.text, node_a.text
```

Comment:
This swapping code block looks like it would be useful to abstract into a class method on the node model.

Reply:
Including the saves here, no need to bloat the controller with logic that makes sense in the model.